src/main/java/com/aws/greengrass/mqttbridge/MessageBridge.java

```diff
@@ -51,8 +52,16 @@
      */
     public MessageBridge(TopicMapping topicMapping) {
         this.topicMapping = topicMapping;
-        this.topicMapping.listenToUpdates(this::processMappingAndSubscribe);
-        processMappingAndSubscribe();
+    }
+
+    public void start() {
+        topicMapping.listenToUpdates(updateListener);
+        updateListener.onUpdate();
```

Comment:
I like calling `processMappingAndSubscribe();` directly better. Because then I don't have to go check what `onUpdate` is doing. The code should read like english as much as possible IMO.

Reply:
agreed, will adjust.   unfortunately for `topicMapping.listenToUpdates(updateListener)` I have to still use `updateListener` so that the proper reference is used when calling `TopicMapping#unsubscribe`